java/algorithm/src/main/java/org/apache/arrow/algorithm/search/SortedVectorSearcher.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.search;
+
+import org.apache.arrow.algorithm.sort.VectorValueComparator;
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Searcher for sorted vectors.
+ * @param <V> vector type.
+ */
+public class SortedVectorSearcher<V extends ValueVector> extends VectorSearcher<V> {
+
+  /**
+   * Constructor for sorted vector searcher.
+   *
+   * @param targetVector the vector to search for.
+   * @param comparator   the criteria to determine vector data equality.
+   */
+  public SortedVectorSearcher(V targetVector, VectorValueComparator<V> comparator) {
+    super(targetVector, comparator);
+  }
+
+  @Override
+  public int search(V keyVector, int keyIndex) {
+    comparator.attachVectors(keyVector, targetVector);
+
+    // perform binary search
+    int low = 0;
+    int high = targetVector.getValueCount() - 1;
+
+    while (low <= high) {
```

Comment:
Does this work on single element vectors?

Reply:
Sure. It works for vectors with arbitrary length:

For a single element vector, low = high = mid = 0;
If it misses in the first iteration, the index (0) will be returned. Otherwise, it will quit the iteration, after updating low or high, and returns -1 afterward.